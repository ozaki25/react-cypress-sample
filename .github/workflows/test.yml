name: Test
on: [pull_request]
jobs:
  unit-test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: Use Node.js 12
        uses: actions/setup-node@v1
        with:
          node-version: 12
      - name: install
        run: |
          yarn install
      - name: lint
        run: |
          yarn lint
      - name: test
        run: |
          yarn test
  e2e-test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: Use Node.js 12
        uses: actions/setup-node@v1
        with:
          node-version: 12
      - name: install
        run: |
          yarn install
      - name: e2e-test
        run: |
          yarn e2e
