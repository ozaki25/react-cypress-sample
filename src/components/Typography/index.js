import React from 'react';
import styled from 'styled-components';

const P = styled.p`
  font-size: 1.2em;
  font-weight: 600;
  margin: 0 0 15px;
  color: ${({ error }) => (error ? '#d32f2f' : '#000')};
`;

function Typography({ error, ...rest }) {
  return <P error={error ? 1 : 0} {...rest} />;
}

export default Typography;
