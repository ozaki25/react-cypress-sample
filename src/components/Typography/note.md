- 通常の文字列を表示するのに使うコンポーネントです
- `error`属性をつけるとエラーを表す赤字になります