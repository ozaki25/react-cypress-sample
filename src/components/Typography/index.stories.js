import React from 'react';
import Typography from '.';
import note from './note.md';

export default {
  title: 'Typography',
  parameters: {
    info: {
      text: note,
    },
  },
};

export const nomal = () => <Typography>通常の文章です</Typography>;

export const error = () => <Typography error>エラーを表す文章です</Typography>;
