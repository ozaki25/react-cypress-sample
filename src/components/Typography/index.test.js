import React from 'react';
import { snapshot } from '../../test/helper';
import Typography from '.';

snapshot('Typography', <Typography>通常の文章です</Typography>);
