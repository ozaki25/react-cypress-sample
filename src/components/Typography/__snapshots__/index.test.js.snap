// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: Typography 1`] = `
.c0 {
  font-size: 1.2em;
  font-weight: 600;
  margin: 0 0 15px;
  color: #000;
}

<p
  className="c0"
>
  通常の文章です
</p>
`;

exports[`should match snapshot: Typography error text 1`] = `
.c0 {
  font-size: 1.2em;
  font-weight: 600;
  margin: 0 0 15px;
  color: #d32f2f;
}

<p
  className="c0"
>
  エラーの文章です
</p>
`;
